### ISSUE TITLE
Build instructions required

### ISSUE THREAD

**Issue by Anthchirp** [nexusformat/HDF5-External-Filter-Plugins#8]
The [xkcd install script](https://xkcd.com/1654/) doesn't work and there are no other helpful pointers.

How do I compile this?

**Comment by eugenwintersberger** [comment]
On which platform?

**Comment by ndevenish** [comment]
Admittedly the README is out-of-date about autotools, and if you aren't familiar with CMake the fact that *everything* is disabled by default isn't that obvious

**Comment by eugenwintersberger** [comment]
Yeah. The README is definitely out of date. And you are right about the CMake file ;). Still what platform are you planning to build on?

**Comment by Anthchirp** [comment]
We want to run this as part of the DIALS installation, so need to cover Linux and MacOS as a minimum.  We need bitshuffle and LZ4, and thanks to @ndevenish I got as far as
```bash
mkdir build
cd build
cmake .. -DENABLE_BITSHUFFLE_PLUGIN=yes -DENABLE_LZ4_PLUGIN=yes -DCMAKE_INSTALL_PREFIX=$somepath
make install
```

however we have to support RHEL6 which means
```
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.0.0 or higher is required.  You are running version 2.8.12.2
```
is an issue. CentOS5 may also on the list of things to support, don't even know what happens on there.

Any chance CMake 3 could be un-required?

**Comment by eugenwintersberger** [comment]
The decision to set the required CMake Version to >=3 was simply due to the far better dependency management in CMake 3. So for me going back to CMake 2 is not an option. But if someone is willing to do the work he or she is mostly welcome. 

When working on legacy systems, and RHEL 5 and 6 are definitely legacy, you may consider using [conan](https://www.conan.io/) to manage build tool dependencies. 

I am currently pretty busy with work so I am not sure when I will have time to update the build instructions. But, again, I definitely should do that.

**Comment by MarkRivers** [comment]
I am using RHEL 7, which I don't consider legacy, and it has cmake 2.8.12.2, so I can build on it either.
